packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -90,28 +85,16 @@ export default function UnsavedInnerBlocks( {
 		}
 	);
 
-	const { isSaving, hasResolvedDraftNavigationMenus } = useSelect(
+	const { isSaving } = useSelect(
 		( select ) => {
 			if ( isDisabled ) {
 				return EMPTY_OBJECT;
 			}
 
-			const {
-				getEntityRecords,
-				hasFinishedResolution,
-				isSavingEntityRecord,
-			} = select( coreStore );
+			const { isSavingEntityRecord } = select( coreStore );
 
 			return {
 				isSaving: isSavingEntityRecord( 'postType', 'wp_navigation' ),
-				draftNavigationMenus: getEntityRecords(
-					// This is needed so that hasResolvedDraftNavigationMenus gives the correct status.
-					...DRAFT_MENU_PARAMS
-				),
-				hasResolvedDraftNavigationMenus: hasFinishedResolution(
-					'getEntityRecords',
-					DRAFT_MENU_PARAMS
-				),
```

Comment:
There was obviously some need for this check to be in place. Therefore I'd be hesitant to remove it without knowing what that reason was.

You are essentially relying on the other request for both public and draft to have resolved.

Perhaps you could instead make this the exact same request as the other one and that way the built in caching layer of Core Data will simply resolve the data for which ever one resolves first. That will still leave you with a single request but will have the safe guard here preserved.

Reply:
This component is already relying on `hasResolvedNavigationMenus` so that seems unnecessary to me.

Reply:
So `hasResolvedNavigationMenus` is _currently_ based on `public` or `draft` status. If that changed to just `public` again in the future then this code _could_ break because of that.

I found the original PR that introduced this https://github.com/WordPress/gutenberg/pull/36024.

It was using `draft` because the main `hasResolvedNavigationMenus` query was only looking at `publish` so it needed an extra query.

Also I would look at why all the tests are suddenly failing on this PR. What's caused that?